### PR TITLE
feat: allow config from CHARM_CONFIG_HOME and XDG_CONFIG_HOME

### LINKED ISSUE #328: Support XDG paths on macOS
It'd be great if glow would read its config from `XDG_CONFIG_HOME` instead of `~/Library/Preferences/glow/glow.yml` on macOS.

I frankly don't know anyone who uses `~/Library` for any kind of configuration for CLI tools.

### LINKED ISSUE #475: Respect XDG Base Directory specification on macOS
I was trying out Glow because of a user-reported issue on Ranger earlier and noticed that Glow defaults to writing configuration to `~/Library` on macOS.

While the Freedesktop [XDG Base Directory specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) isn't really targeted at macOS, so many programs do respect the specification and make it a lot more convenient to manage configurations and such.

Could it be considered that Glow use the XDG Base Directories if the variables are defined in the environment and fall back to the usual location in `~/Library` otherwise?

Another minor note is that I was surprised Glow unconditionally writes a config file when running the `glow config` subcommand, even if no settings are changed. In general I prefer to configure only the things I specifically care about and fall back to defaults, which may change over time, determined to be best by upstream. Maybe the subcommand could check if any changes were made and only write the config if that was the case? And maybe it should offer all the possible settings commented out so you can set just the ones you care about and fall back to defaults for everything else?

### LINKED ISSUE #495: Support `GLOW_CONFIG_DIR` env var
hi 👋🏼 

I'm trying to consolidate my dotfiles to be the same on macos and linux and have a small issue with how glow handles its config file.

I have a script that runs `yq --inplace '.style = "dark"' ~/.config/glow/glow.yml` when switching between dark/light in my OS to set the right theme for glow but can't reuse the same script on mac and linux  because on mac the path is `~/Library/Preferences/glow/glow.yml` vs `~/.config/glow/glow.yml` on linux.
[I saw this issue requesting to support XDG paths](https://github.com/charmbracelet/glow/issues/328) and then found that glow uses `go-app-paths` to handle platform-specific paths so this should be fixed/added there to work on mac (XDG is supported on linux).

This feature request is to allow users to specify a path for the config file via an env var like `GLOW_CONFIG_DIR`.
It wouldn't be a breaking change like adding support for XDG  on macos and seems easy to add (happy to open a PR  if you'll agree on this request :) )

### PR DESCRIPTION
**Pull request by caarlos0**
closes #594
closes #495
closes #328
closes #475

### CODE REVIEW

File: config_cmd.go [hunk 1]
```diff
@@ -23,6 +23,12 @@ width: 80
 
 func defaultConfigFile() string {
 	scope := gap.NewScope(gap.User, "glow")
+	if c := os.Getenv("CHARM_CONFIG_HOME"); c != "" {
+		return filepath.Join(c, "glow.yml")
+	}
+	if c := os.Getenv("XDG_CONFIG_HOME"); c != "" {
```
Comment (ccoVeille) [inline config_cmd.go:29]: Why not using os.UseConfigDir ?

https://pkg.go.dev/os#UserConfigDir

It's implementation might be what you are looking for

https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/os/file.go;l=509

Comment (Delta456) [reply config_cmd.go:29]: UserConfigDir also looks at the home dir `os.GetEnv("HOME")`  if `os.GetEnv("XDG_CONFIG_HOME")` is empty which isn't wanted here.

Comment (ccoVeille) [reply config_cmd.go:29]: Indeed

Comment (caarlos0) [reply config_cmd.go:29]: > On Darwin, it returns $HOME/Library/Application Support

I think it wouldn't change much in that particular problem.

We can work on replacing the usage of gap with this library though

Comment (ccoVeille) [reply config_cmd.go:29]: Please note, I know some macOS users expect to use this folder by default, but want code to respect XDG_CONFIG_HOME if present.

- https://github.com/charmbracelet/glow/pull/594#issuecomment-2189627733

Comment (caarlos0) [reply config_cmd.go:29]: yes, this is what this PR does...

File: main.go [hunk 2]
```diff
@@ -388,13 +388,22 @@ func init() {
 }
 
 func tryLoadConfigFromDefaultPlaces() {
+	fmt.Println("AQUI 1")
```
Comment (ccoVeille) [inline main.go:391]: This one was forgotten

File: main.go [hunk 3]
```diff
@@ -388,13 +388,22 @@ func init() {
...
 		fmt.Println("Could not load find config directory.")
 		os.Exit(1)
 	}
 
+	if c := os.Getenv("CHARM_CONFIG_HOME"); c != "" {
+		viper.AddConfigPath(c)
+	}
+
+	if c := os.Getenv("XDG_CONFIG_HOME"); c != "" {
+		viper.AddConfigPath(filepath.Join(c, "glow"))
```
[the remark this answers is not in the document]
Comment (caarlos0) [reply main.go:403]: it should respect order yet

**Pull request merged**